### PR TITLE
Add UC-1 keypad

### PR DESCRIPTION
**Pull request by duoshock**
UC-1 is a 3 keys macro pad with encoder.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/merge/uc1/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/* key matrix size */
+#define MATRIX_ROWS 2
+#define MATRIX_COLS 2
+
+#define MATRIX_ROW_PINS { B1, B2 }
+#define MATRIX_COL_PINS { B3, B4 }
+#define UNUSED_PINS
+
+#define USB_MAX_POWER_CONSUMPTION 500
+
```
Comment (zvecr) [inline keyboards/merge/uc1/config.h:34]:
```suggestion

```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/merge/uc1/config.h (hunk 1 above)
Comment (duoshock) [reply keyboards/merge/uc1/config.h]: Thanks. I've updated it.

File: keyboards/merge/uc1/config.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  */ 
+
+#include "config_common.h"
+
+#define VENDOR_ID       0x4d65
```
Comment (drashna) [inline keyboards/merge/uc1/config.h:19]:
```suggestion
#define VENDOR_ID       0x4D65
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/merge/uc1/config.h (hunk 2 above)
Comment (duoshock) [reply keyboards/merge/uc1/config.h]: Thanks for the feedback. It's updated.

**Review by drashna — APPROVED** [review]

File: keyboards/merge/uc1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        	# Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    	# Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # Nkey Rollover 
+BACKLIGHT_ENABLE = no      	# Enable keyboard backlight functionality
```
Comment (fauxpark) [inline keyboards/merge/uc1/rules.mk:19]:
```suggestion
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # Nkey Rollover 
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/merge/uc1/rules.mk (hunk 3 above)
Comment (duoshock) [reply keyboards/merge/uc1/rules.mk]: It's updated.

File: keyboards/merge/uc1/rules.mk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = no			# Console for debug
+COMMAND_ENABLE = no			# Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no		# Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no			# Nkey Rollover 
+BACKLIGHT_ENABLE = no		# Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes		# Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no		# Enable Bluetooth
+AUDIO_ENABLE = no			# Audio output
```
Comment (fauxpark) [inline keyboards/merge/uc1/rules.mk:22]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # Nkey Rollover 
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
Also missing an info.json.

**Comment by duoshock** [comment]
> Also missing an info.json.

Sorry. It's added.

File: keyboards/merge/uc1/info.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "keyboard_name": "UC-1",
+  "url": "https://mergedesign.store/products/uc-1",
+  "maintainer": "duoshock",
+  "width": 4.25,
+  "height": 1,
+  "layouts": {
+    "LAYOUT_all": {
+      "key_count": 4,
```
Comment (fauxpark) [inline keyboards/merge/uc1/info.json:9]:
```suggestion
```

File: keyboards/merge/uc1/info.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{
+  "keyboard_name": "UC-1",
+  "url": "https://mergedesign.store/products/uc-1",
+  "maintainer": "duoshock",
+  "width": 4.25,
+  "height": 1,
+  "layouts": {
+    "LAYOUT_all": {
```
Comment (fauxpark) [inline keyboards/merge/uc1/info.json:8]:
```suggestion
    "LAYOUT": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by duoshock** [comment]
Hello, Can you review the changes, please?
I'm not proficient in Github if there's something missing.
Thanks!

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**